cpu/atmega_common/periph/i2c.c

```diff
@@ -123,16 +123,20 @@ int i2c_init_master(i2c_t dev, i2c_speed_t speed)
 
 int i2c_acquire(i2c_t dev)
 {
-    assert(dev < I2C_NUMOF);
-    mutex_lock(&lock);
-    return 0;
+    if (dev < I2C_NUMOF) {
```

Comment:
Here and below: I think there is a reason why the author chose `assert()` over `if ()`. Better use `(void) dev` instead.

Reply:
There seem to be the general consent that `if ()` should be used instead of `assert()`, at least most platforms do, accept this and the `nrf51`. Maybe someone more into periph can give there opinion here, @gebart or @haukepetersen?

But if the change-over to `if ()` should happen I would rather like to see it in a separate PR and also for `nrf51` as well.

Reply:
the reason for that is, that the API requires to return an error for invalid device, so an `assert` is just not enough here

Reply:
shall I also factor this one out into a separate PR?

Reply:
I'd prefer that.